pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -303,7 +303,9 @@ private <C extends IndexConfig> ColumnConfigDeserializer<C> getDeserializer(Inde
         deserializer = IndexConfigDeserializer.fromMap(table -> fromIndexLoadingConfig);
       } else if (_segmentTier == null) {
         deserializer =
-            IndexConfigDeserializer.fromMap(table -> fromIndexLoadingConfig).withFallbackAlternative(stdDeserializer);
+            stdDeserializer.withFallbackAlternative(IndexConfigDeserializer.fromMap(table -> fromIndexLoadingConfig));
+//        deserializer =
+//            IndexConfigDeserializer.fromMap(table -> fromIndexLoadingConfig).withFallbackAlternative(stdDeserializer);
```

Comment:
I guess we can remove the comment, Am I right?

Reply:
Moved this change to a separate PR -> https://github.com/apache/pinot/pull/12265